### PR TITLE
Feature/dependency injection

### PR DESCRIPTION
**Pull request by iampeterbanjo**
addresses #31

### CODE REVIEW

**Comment by SimonWaldherr** [comment]
thanks for your pr, can you please add your example to the readme file.
what should `go Shellrun` do??

File: readme.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
+These examples explain the basics of golang. There will be more examples from time to time.
 
 if you like, feel free to add more golang examples. Many thanks to all [contributors](https://github.com/SimonWaldherr/golang-examples/graphs/contributors).
 
 ## Install go(lang)
 
 with [homebrew](http://mxcl.github.io/homebrew/):
 
-```
+```go
```
Comment (SimonWaldherr) [inline readme.md]: `sudo brew install go` is not go-code

File: readme.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 sudo brew install go
 ```
 
 with [apt](http://packages.qa.debian.org/a/apt.html)-get:
 
-```
+```go
```
Comment (SimonWaldherr) [inline readme.md]: again, `apt-get install golang` is also not golang code

File: readme.md [hunk 3]
```diff
@@ -57,394 +57,401 @@ If you know ```#!```, also known as [Shebang](https://en.wikipedia.org/wiki/Sheb
 
 Print Hello World with comments
 
-```
+```go
```
Comment (SimonWaldherr) [inline readme.md]: not go code

File: readme.md [hunk 4]
```diff
@@ -57,394 +57,401 @@ If you know ```#!```, also known as [Shebang](https://en.wikipedia.org/wiki/Sheb
...
 go run HelloWorld.go
 ```
 
 Print Hello World with comments (shebang version)
 
-```
+```go
```
Comment (SimonWaldherr) [inline readme.md]: not go code

File: readme.md [hunk 5]
```diff
@@ -57,394 +57,401 @@ If you know ```#!```, also known as [Shebang](https://en.wikipedia.org/wiki/Sheb
...
 ./HelloWorldShebang.go
 ```
 
 Declare variables and print them
 
-```
+```go
```
Comment (SimonWaldherr) [inline readme.md]: not go code

File: readme.md [hunk 6]
```diff
@@ -57,394 +57,401 @@ If you know ```#!```, also known as [Shebang](https://en.wikipedia.org/wiki/Sheb
...
 ### Expert
 
 Calculate π with go (leibniz, euler and prime are running until you stop it via CTRL+C)
 
-```
-go run pi2go.go leibniz
-go run pi2go.go euler
+```go
+go Shellrun pi2go.go leibniz
+go Shellrun pi2go.go euler
```
Comment (SimonWaldherr) [inline readme.md]: `go Shellrun`?

Comment (iampeterbanjo) [reply readme.md]: fixed. sorry i over did the 'find and replace' (-_-*)

File: readme.md [hunk 7]
```diff
@@ -57,394 +57,401 @@ If you know ```#!```, also known as [Shebang](https://en.wikipedia.org/wiki/Sheb
...
+```Shell
 go run image.go
 ```
 
 Sql (sqlite) golang example
 
-```
-go run sqlite.go insert test
+```Shell
+go Shellrun sqlite.go insert test
```
Comment (SimonWaldherr) [inline readme.md]: still one `Shellrun`

**Comment by SimonWaldherr** [comment]
Please think about whether you want to consider a rebase to cleanup the commit history: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

File: readme.md (hunk 7 above)
Comment (iampeterbanjo) [reply readme.md]: fixed

**Comment by iampeterbanjo** [comment]
thanks. i rebased it

**Pull request merged**